File: javaFile/insert_at_position.java
class Node {
    int data;
    Node next;
    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;
    LinkedList() {
        this.head = null;
    }

    void createList(int data){
        Node newNode = new Node( data);
        if (head == null){
            head = newNode;
            return;
        }
        Node current = head;
        while  (current.next != null) {
            current = current.next;
        }
        current.next = newNode;

    }

    void insertAtPosition(int position, int data) {
        Node newNode = new Node(data);
        if (position == 1) {
            newNode.next = head;
            head = newNode;
        }
        Node current = head;
        for(int i=1; i<position; i++) {
            if (current == null) {
                System.out.println("Position out of bounds");
                return;
            }
            current = current.next;
        }
        newNode.next = current.next;
        current.next = newNode;

    }
    void printList() {
        Node temp = head;
        while ( temp != null) {
            System.out.println(temp.data+ "-> ");
            temp = temp.next;
        }
    }


}

public class insert_at_position {

    public static void main(String[] args) {
        LinkedList link_list = new LinkedList();
        link_list.createList(10);
        link_list.createList(20);
        link_list.createList(30);
        System.out.println("befor inserting at position");
        link_list.printList();
        
        int position = 3;
        int data = 25;
        link_list.insertAtPosition(position, data);
        System.out.println("after inserting data at position");
        link_list.printList();
        
    }
}

File: python/insertAtPosition.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head= None
    
    def create_list(self, data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
            return
        current = self.head
        while current.next:
            current = current.next
        current.next = new_node

    def insertATPosition(self, position, data):
        new_node = Node(data)
        if position == 1:
            new_node.next = self.head
            self.head = new_node
            return
        current = self.head
        for _ in range(1,position - 1):
            if current is None:
                print("Position out of bounds")
                return
            current = current.next
    
        new_node.next = current.next
        current.next = new_node
    
    def print_list(self):
        current = self.head
        while current:
            print(current.data, end=" -> ")
            current = current.next
        print("None")


def main():
    linked_list = LinkedList()
    linked_list.create_list(1)
    linked_list.create_list(2)
    linked_list.create_list(3)
    linked_list.create_list(4)
    linked_list.create_list(5)

    print("Original List:")
    linked_list.print_list()

    position = 3
    data = 10
    linked_list.insertATPosition(position, data)

    print(f"List after inserting {data} at position {position}:")
    linked_list.print_list()

if __name__ == "__main__":
    main()
        

        